docs/concepts/ale/ale-driver-development-concepts.md

```diff
@@ -0,0 +1,41 @@
+# ALE Driver Development Concepts
```

Comment:
maybe call this like "Architecture of Ale Drivers"? Open to other ideas but this is a bit odd.